### DISCUSSION TITLE
Faster PostgreSQL query times

### DISCUSSION THREAD

**Discussion by konotorii** [prisma/prisma#16878] [Q&A]
I've got a database with over 54 Million rows and I have to constantly query for about 100 of these rows that contain data about a score made in a game for a specific user. I've tried researching ways to increase this speed but have come unfruitful. I ran ``explain analyze`` and the Execution time is ``14230.548 ms`` and Planning time is ``4.127 ms``. How do I reduce the Execution time to something more high performant and less time consuming?

**Comment by MwSpaceLLC** [comment]
Che tipo di query stai utilizzando?

**Reply by konotorii** [reply]
findMany where user_id = {ID im searching for}

**Comment by pantharshit00** [comment]
Hello @konotorii,

Can you please share relevant model and the client call which you are using? Right now it is hard to assess the problem without this information.

**Comment by konotorii** [comment]
hi sorry for late response.

my model:
``model score {
  score_id        String   @unique
  user_id         Int
  beatmapId       Int
  mod             String
  mod_enum        Int
  score           String
  maxcombo        Int
  rank            String
  count50         Int
  count100        Int
  count300        Int
  countmiss       Int
  countgeki       Int
  countkatu       Int
  perfect         Int
  date            DateTime
  pp              Float
  replay          Int
  hidden          Int?
  total_length    Int?
  hit_length      Int?
  country_acronym String?
}``

client call: 
``const topplays = await this.prisma.score.findMany({
            where: {
                user_id: user.osu_id
            },
            orderBy: {
                pp: 'desc'
            },
            take: 100
        })``
